Kernel/FileSystem/SysFS/Subsystems/Kernel/Variables/FixedStringBufferVariable.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/FileSystem/SysFS/Subsystems/Kernel/Variables/FixedStringBufferVariable.h>
+#include <Kernel/Sections.h>
+#include <Kernel/Tasks/Process.h>
+
+namespace Kernel {
+
+ErrorOr<void> SysFSSystemFixedStringBufferVariable::try_generate(KBufferBuilder& builder)
+{
+    auto string_value = TRY(value());
+    return builder.appendff("{}\n", string_value->view());
+}
+
+ErrorOr<size_t> SysFSSystemFixedStringBufferVariable::write_bytes(off_t offset, size_t count, UserOrKernelBuffer const& buffer, OpenFileDescription*)
+{
+    if (offset != 0)
+        return EINVAL;
+    if (m_storage_during_write_buffer_size < count)
+        return E2BIG;
+    // NOTE: If we are in a jail, don't let the current process to change the variable.
+    if (Process::current().is_currently_in_jail())
+        return Error::from_errno(EPERM);
+    MutexLocker locker(m_refresh_lock);
+    TRY(m_storage_during_write.with([&](auto& storage) -> ErrorOr<void> {
+        storage.fill_with(0);
+        TRY(buffer.read(storage.span().data(), count));
+        auto new_value_without_possible_newlines = StringView(storage.span()).trim("\n"sv);
```

Comment:
What's the goal here? Do we accept newlines in the middle of the hostname?

Reply:
You could send a new line inside a string to a sysfs string variable file. We should truncate them (both at the end and in the middle) :)

Reply:
Maybe instead of truncation, we should just replace new lines with spaces? I don’t know if that will look weird though, but still sending new lines to variables that should be just one line is quite odd as well.

Reply:
Maybe it should just be a valid hostname, and the write could fail if it's not?